mri/scripts/gridsearch.py

```diff
@@ -0,0 +1,188 @@
+# -*- coding: utf-8 -*-
+##########################################################################
+# pySAP - Copyright (C) CEA, 2017 - 2018
+# Distributed under the terms of the CeCILL-B license, as published by
+# the CEA-CNRS-INRIA. Refer to the LICENSE file or to
+# http://www.cecill.info/licences/Licence_CeCILL-B_V1-en.html
+# for details.
+##########################################################################
+
+import itertools
+
+from joblib import Parallel, delayed
+import numpy as np
+
+
+def _reconstruct_case(case, key_names, init_classes,
+                      kspace_data, fourier_kwargs):
+    """Internal Function to carry out reconstruction for a
+    special case. This function pulls in appropriate keyword arguments
+    from input and declares appropriate Linear, Fourier and Regularizer
+    Operators. These operators later are used to create the image model
+    by defining the reconstructor. Then the reconstruction is carried and
+    results are returned
+
+    Parameters
+    ----------
+    case: list of parameters
+        keyword arguments for different operators for reconstruction
+    key_names: list of strings
+        names of keyword arguments specified for reconstruction in 'case'
+    init_classes: list of classes
+        classes for varoius operators. must be in format
+        [linear, regularizer, reconstructor]
+    kspace_data: np.ndarray
+        the kspace data for reconstruction
+    fourier_kwargs: dict
+        holds dictionary with init_class pointing to fourier class to be used
+        and args having keyword arguments for initialization
+        NOTE: We declare fourier operator inside this function to allow
+        parallel execution as NonCartesianFFT cannot be pickled.
```

Comment:
Since NonCartesianFFT, particularly NFFT uses cython, it had pickling errors:

`TypeError: no default __reduce__ due to non-trivial __cinit__`

However, there are some solutions I could find https://stackoverflow.com/questions/12646436/pickle-cython-class, but I didnt give it a try as they mostly would br PRs to pyNFFT

Reply:
couldn't we have a switch instead ? this is really cumbersome, and the user shouldn't have to define the fourier op twice, which right now is the case.

Reply:
What do you mean by switch? 
Even if we used a swtich statement to define NFFT or Cartesian, we would still need to pass all the parameters twice right?

Reply:
Added both fouier_op and fourier_params. Example tests with fourier_op and `FFT` and test with fourier_params and `NonCartesianFFT`.
Also I opened issue in pyNFFT about adding __reduce__ : https://github.com/pyNFFT/pyNFFT/issues/71